centrosome/zernike.py

```diff
@@ -40,36 +46,58 @@ def construct_zernike_polynomials(x, y, zernike_indexes, mask=None, weight=None)
     if x.shape != y.shape:
         raise ValueError("X and Y must have the same shape")
     if mask is None:
-        mask = np.ones(x.shape,bool)
+        no_mask = True
     elif mask.shape != x.shape:
         raise ValueError("The mask must have the same shape as X and Y")
-    x = x[mask]
-    y = y[mask]
-    if weight is not None:
-        weight = weight[mask]
-    lut = construct_zernike_lookuptable(zernike_indexes)
+    else:
+        no_mask = False
+        x = x[mask]
+        y = y[mask]
+        if weight is not None:
+            weight = weight[mask]
+    lut = construct_zernike_lookuptable(zernike_indexes) # precompute poly. coeffs.
     nzernikes = zernike_indexes.shape[0]
-    r = np.sqrt(x**2+y**2)
-    phi = np.arctan2(x,y).astype(np.complex)
-    zf = np.zeros((x.shape[0], nzernikes), np.complex)
-    s = np.zeros(x.shape,np.complex)
-    exp_terms = {}
+    # compute radii
+    r_square = np.square(x) # r_square = x**2
+    np.add(r_square, np.square(y), out=r_square) # r_square = x**2 + y**2
```

Comment:
yes, but uses 2 intermediate arrays rather than 3 as in `np.add(np.square(x), np.square(y))`, one for each square, and one for the result of `add`.